src/stan/mcmc/hmc/hamiltonians/diag_e_metric.hpp

```diff
@@ -30,19 +30,27 @@ namespace stan {
         return this->V(z);
       }
 
-      double dG_dt(diag_e_point& z) {
+      double dG_dt(diag_e_point& z,
+                   interface_callbacks::writer::base_writer& info_writer,
+                   interface_callbacks::writer::base_writer& error_writer) {
         return 2 * T(z) - z.q.dot(z.g);
       }
 
-      const Eigen::VectorXd dtau_dq(diag_e_point& z) {
+      const Eigen::VectorXd dtau_dq(
+        diag_e_point& z,
+        interface_callbacks::writer::base_writer& info_writer,
+        interface_callbacks::writer::base_writer& error_writer) {
         return Eigen::VectorXd::Zero(this->model_.num_params_r());
       }
 
       const Eigen::VectorXd dtau_dp(diag_e_point& z) {
```

Comment:
(questions) 

1.  argument diag_e_point& should be const as it's not modified (though you may have problems with `z.p`;  if we cared more about const correctness, you'd use a const method to return `z.p`.

2.  I don't think the const on the return matters because it's a copy.

3. If this gets used to assign, it might be best to pass that in so that there's not a redundant copy on the `cwiseProduct()` return.